File: 87 3-D Plot/ruby/3dplot.rb
def intro
  puts "                                3D PLOT
               CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n\n\n\n\n"
end

def fna(z) = 30 * Math.exp(-z * z / 100)

def render
  (-30..30).step(1.5).each do |x|
    l = 0
    y1 = 5 * (Math.sqrt(900 - x * x) / 5).to_i
    y_plot = " " * 80
    (y1..-y1).step(-5).each do |y|
      z = (25 + fna(Math.sqrt(x * x + y * y)) - 0.7 * y).to_i
      next if z <= l
      l = z
      y_plot[z] = '*'
    end
    puts y_plot
  end
end

def main
  intro
  render
end

main

File: 87 3-D Plot/perl/3dplot.pl
#!/usr/bin/perl
use strict;
use warnings;

print ' 'x32 ."3D PLOT\n";
print ' 'x15 ."CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n";
print "\n\n\n";

sub FNA {
	my ($Z)= @_;
	return 30*exp(-$Z*$Z/100);
	}

print "\n";

for (my $X=-30; $X<=30; $X+=1.5) {
	my $L=0;
	my $Line=" "x80; #Empty buffer string;
	my $Y1=5*int(sqrt(900-$X*$X)/5);
	for (my $Y=$Y1; $Y>=-$Y1; $Y-=5) {
		my $Z=int(25+&FNA(sqrt($X*$X+$Y*$Y))-.7*$Y);
		if ($Z<=$L) { next; }
		$L= $Z;
		substr $Line, $Z, 1, "*"; #Plot on the line by sustitution.
		}
	print "$Line\n"; #Now print the line.
	}
